opendatahub-io/odh-manifests
# Configure Superset to use OpenShift OAuth

This change:

1. Configures Superset to use OpenShift OAuth for its identity provider
2. Switches the superset route to use SSL
3. Makes it easier to extend the default superset configuration set in superset_config.py

This addresses https://issues.redhat.com/browse/ODH-566

To test this:

1. Run the following commands:
```
oc new-project opendatahub
kustomize build superset/base/ | oc apply -f -
```
2. Navigate to Superset (get the URL from the Superset route)
3. Click the login button in the top right of Superset
4. Log in with OpenShift
5. Verify that you are a Superset admin (you should be able to see admin functionality (e.g. list users, roles) when you hover over the settings menu in the top right)
6. Have someone who isn't a project admin in your openshift cluster test access and confirm that they are not a superset admin
